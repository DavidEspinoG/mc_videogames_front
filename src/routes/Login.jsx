import { BiArrowBack } from 'react-icons/bi';
import '../styles/Login.scss';
import { useNavigate } from 'react-router-dom';

const Login = () => {
  const navigate = useNavigate();
  return (
    <div className="login-main-container">
      <div className="container">
        <div className="login-header">
          <BiArrowBack className="back-icon" onClick={() => navigate('/')} />
        </div>
        <div className="login-form">
          <h3 className="subtitle text-center">LOGIN</h3>
          <hr className="green-line" />
          <div className="login-inputs">
            <input placeholder="Email" className="input" />
            <input placeholder="Password" className="input" />
            <button type="button" className="login-submit-button">
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login;
