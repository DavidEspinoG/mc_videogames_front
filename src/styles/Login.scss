.login-main-container {
  background-image: url('../assets/green_background.png');
  background-position: center;
  background-size: cover;
  height: 100vh;
  font-family: var(--main-font);
}

.container {
  width: 100%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.back-icon {
  color: white;
  font-size: 25px;
}

.back-icon:hover {
  cursor: pointer;
}

.login-icon-two {
  color: var(--green);
  background-color: white;
  padding: 13px;
  border-radius: 40%;
}

.login-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
}

.subtitle {
  font-weight: 700;
  letter-spacing: 1.5px;
  margin: 0;
}

.green-line {
  border: 1px solid #bde244;
  width: 600px;
  max-width: 80%;
}

.login-form {
  color: white;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.input {
  color: white;
  padding: 20px;
  border-radius: 30px;
  border: 1px solid white;
  background-color: var(--green);
  width: 270px;
}

.input::placeholder {
  color: white;
}

.input:focus {
  outline: none;
  transform: scale(1.02);
}

.login-inputs {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 10px;
  margin-bottom: 15px;
  max-width: 90%;
}

.text-center {
  text-align: center;
}

.login-submit-button {
  background-color: white;
  color: var(--green);
  padding: 20px;
  border: none;
  border-radius: 30px;
  width: 200px;
}

.underline {
  text-decoration: underline;
}

.login-submit-button:hover {
  transform: scale(1.1);
}

@media (min-width: 768px) {
  .login-inputs {
    flex-direction: row;
  }

  .input {
    width: 200px;
  }
}
