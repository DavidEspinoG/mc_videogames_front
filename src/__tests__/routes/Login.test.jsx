import Login from '../../routes/Login';
import renderWithProviders from '../test-utils';

describe('Login', () => {
  it('should render as expected', () => {
    const { container } = renderWithProviders(<Login />);
    expect(container).toMatchSnapshot();
  });
});
