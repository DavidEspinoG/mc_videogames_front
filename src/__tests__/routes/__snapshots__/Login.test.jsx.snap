// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > should render as expected 1`] = `
<div>
  <div
    class="login-main-container"
  >
    <div
      class="container"
    >
      <div
        class="login-header"
      >
        <svg
          class="back-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
          />
        </svg>
      </div>
      <div
        class="login-form"
      >
        <h3
          class="subtitle text-center"
        >
          LOGIN
        </h3>
        <hr
          class="green-line"
        />
        <form
          class="login-inputs"
        >
          <input
            class="input"
            placeholder="Email"
            type="text"
            value=""
          />
          <input
            class="input"
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            class="login-submit-button"
            type="submit"
          >
            Submit
          </button>
        </form>
        <div>
          <span>
            Don't have an account? 
          </span>
          <a
            class="underline"
            href="/signin"
          >
            Create an account
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
